syncstorage/db.go

```diff
@@ -887,7 +887,14 @@ func (d *DB) getBSOs(
 	}
 
 	limitStmt := "LIMIT ?"
-	values = append(values, limit)
+
+	// fetch an extra row to detect if there are more
+	// rows that match the query conditions
+	if limit > 0 {
+		values = append(values, limit+1)
+	} else {
+		values = append(values, limit)
+	}
```

Comment:
Good point. I suppose a client should be able to send limit=0 to get a 0 byte body but information if there are records actually to download. 

So a limit=0 should return an X-Weave-Next-Offset header.